src/frontend/data/genericReducers/resourceList/actions.ts

```diff
@@ -0,0 +1,65 @@
+import { APIListCommonRequestParams } from '../../../types/api';
+import { modelName } from '../../../types/models';
+import { Resource } from '../../../types/tracks';
+import { Maybe } from '../../../utils/types';
+
+export interface ResourceListGet {
+  jwt: string;
+  params: Partial<APIListCommonRequestParams> & {
+    [key: string]: Maybe<string | number | Array<string | number>>;
+  };
+  resourceName: modelName;
+  type: 'RESOURCE_LIST_GET';
+}
+
+export function getResourceList(
+  jwt: string,
+  resourceName: modelName,
+  params: ResourceListGet['params'] = {},
+): ResourceListGet {
+  return {
+    jwt,
+    params,
+    resourceName,
+    type: 'RESOURCE_LIST_GET',
+  };
+}
+
+export interface ResourceListGetFailure {
+  error: Error | string;
+  resourceName: modelName;
+  type: 'RESOURCE_LIST_GET_FAILURE';
+}
+
+export function failedToGetResourceList(
+  resourceName: modelName,
+  error: ResourceListGetFailure['error'],
+): ResourceListGetFailure {
+  return {
+    error,
+    resourceName,
+    type: 'RESOURCE_LIST_GET_FAILURE',
+  };
+}
+
+export interface ResourceListGetSuccess<R extends Resource> {
+  apiResponse: { objects: R[] };
+  params: Partial<APIListCommonRequestParams> & {
+    [key: string]: Maybe<string | number | Array<string | number>>;
+  };
+  resourceName: modelName;
+  type: 'RESOURCE_LIST_GET_SUCCESS';
+}
+
+export function didGetResourceList<R extends Resource>(
```

Comment:
`getResourceListSuccess` ?

Reply:
I wanted to use a verb in the past tense like in `failedToGetResourceList` to represent the temporal relation to the effect

Reply:
Ok